api/client/webclient/webclient.go

```diff
@@ -428,6 +433,32 @@ type GithubSettings struct {
 	Display string `json:"display"`
 }
 
+// SSHProxyHostPort returns the SSH proxy address for this ping response.
+func (ps *ProxySettings) SSHProxyHostPort() (host, port string, err error) {
+	// If the cluster has TLS routing enabled, SSH proxy is served on the web
+	// handler. Prefer public web address with a fallback to listen address.
+	if ps.TLSRoutingEnabled {
+		if ps.SSH.PublicAddr != "" {
+			return parseHostPortWithDefault(ps.SSH.PublicAddr, defaults.ProxyWebListenPort)
```

Comment:
nit: Should the default port be `ps.SSH.WebListener.Port` if non-empty?

As a user, I'd expect the following config to produce `proxy.example.com:443`, but this would currently produce `proxy.example.com:3080`.

```
proxy_service:
    public_addr: ["proxy.example.com"]
    web_listen_addr: 0.0.0.0:443
```

Note: this is a potential problem across the board that I've noticed, so it's somewhat unrelated to this PR.